### PR TITLE
add drone secrets support for params

### PR DESCRIPTION
**Pull request by svozza**
Adds support for using Drone secrets as cfn params. Summary of the changes:

- Upgraded to latest version of Node
- Upgraded `cfn` but found a bug so have forked and fixed it there (https://github.com/Nordstrom/cfn/pull/26)
- Tidied up some Highland stream stuff
- The negative tests for `resolveParams` don't fail if the error isn't emitted, this has been fixed.
- Optimised the the Dockerfile's build time by installing dependencies before copying over the src files.
- Added documentation and tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Drone plugin for creating/updating or deleting an AWS CloudFormation stack",
   "main": "index.js",
   "scripts": {
-    "pretest": "eslint .",
+
```
Comment (robertstettner) [inline package.json]: Why remove the linting?

**Comment by robertstettner** [comment]
Aren't secrets able to be accessed via environment variables like so:
```
mystep:
  image: myimage
  secrets: [ mysecret ]
  params:
    mycoolparam: $${MYSECRET}
```

File: package.json (hunk 1 above)
Comment (svozza) [reply package.json]: Ooops, not intentional!

**Comment by svozza** [comment]
Really? I was going by this: http://docs.drone.io/secrets-not-working/#variable-expansion:
```
Please note that variable expansion of secrets is not supported. The following yaml configuration will not work.
```
I'll be gutted now if that works because that means this PR is completely useless!

**Comment by robertstettner** [comment]
You might be right, this may only work for commands:
http://docs.drone.io/manage-secrets/#example-in-commands

**Comment by svozza** [comment]
Yeah, I think it's because `commands` is running inside the container so has access to the environment variables.

**Review by robertstettner — APPROVED** [review]

**Review by robertstettner — CHANGES_REQUESTED** [review]
Actually, could you change the `.travis.yml` file to point to version 10 of node?

**Comment by svozza** [comment]
Done!

**Review by robertstettner — APPROVED** [review]

**Pull request merged**